pfcpiface/pfcpiface.go

```diff
@@ -352,3 +354,27 @@ func handleSessionDeletionRequest(upf *upf, msg message.Message, addr net.Addr,
 
 	return smres
 }
+
+
+func initiateAssociationSetupRequest(sourceIP string, n3ip string, n4DstIp string, conn *net.UDPConn) {
+    // Build request message
+    asreq, err := message.NewAssociationSetupRequest(ie.NewRecoveryTimeStamp(time.Now()),
+            ie.NewNodeID(sourceIP, "", ""),       /* node id (IPv4) */
+            // 0x41 = Spare (0) | Assoc Src Inst (1) | Assoc Net Inst (0) | Tied Range (000) | IPV6 (0) | IPV4 (1)
+            //      = 01000001
+            ie.NewUserPlaneIPResourceInformation(0x41, 0, n3ip, "", "", ie.SrcInterfaceAccess),
+            ).Marshal() /* userplane ip resource info */
+        if err != nil {
+            log.Fatalln("Unable to create association setup response", err)
+        }
+
+    smfAddr, err := net.ResolveUDPAddr("udp", n4DstIp+":"+PFCPPort)
+        if err != nil {
+            log.Fatalln("Unable to resolve udp addr!", err)
+            return
+        }
+
+    if _, err := conn.WriteTo(asreq, smfAddr); err != nil {
+        log.Fatalln("Unable to transmit association setup request ", err)
+    }
+}
```

Comment:
Maybe add logic for association request retrials if we don't get the response back from the CP/SMF?? Please see how `cpConnected` is being used in the code as well.

Reply:
Also this should either be called only if SMF IP/name `!= ""`. This way we support both passive and active based on config?